server/socket/index.js

```diff
@@ -0,0 +1,72 @@
+const jwt = require("jsonwebtoken");
+const socket = require("socket.io");
+const socketCookieParser = require("socket.io-cookie-parser");
+
+const onlineUsers = require("../onlineUsers");
+
+exports.appSocket = (server) => {
+  const io = socket(server,{
+    cors: {
+        origin: "http://localhost:3000",
+        credentials: true,
+    }
+  });
+
+  // use Cookie parser
+  io.use(socketCookieParser());
+
+  // user authentication
+  io.use((socket, next) => {
+    const token = socket.request.cookies['token'];
+    if (token === null || !token) {
+        console.log(`There is no access token Found! time =>${new Date().toLocaleString()}`);
+        return next(new Error(`There is no token Found! time =>${new Date().toLocaleString()}`));

```

Comment:
Would there be a way to do error handling when we are passing this error down in middleware so we don't have to have too many console logs?

Reply:
I am sorry. Are you suggesting that we should remove console.log from here or showing the error message on the frontend also when user is not connected with sockets,. I didn't understand it. Please let me know about it.

Reply:
Not at all, just noticed how you are doing `next(new Error(' . . .'))` and was curious if that could be refactored to be in middleware so we can do the logging on the back-end in the middleware then send up the socket response. It would remove all the back-end console logs and move it to just one console log in the middleware. Not a critical task to do right now.

Reply:
That is a better idea. I will that do.